LLMNeedleHaystackTester.py

```diff
@@ -389,14 +336,23 @@ def evaluate_response(self, response):
 
         return int(eval_result['score'])
 
-    def get_context_length_in_tokens(self, context):
-        if self.model_provider == "OpenAI":
-            return len(self.enc.encode(context))
-        elif self.model_provider == "Anthropic":
-            # Assuming you have a different encoder for Anthropic
-            return len(self.enc.encode(context).ids)
+    def evaluate_response_substring_match(self, response):
+        response_lower = response.lower()
+        if all(word in response_lower for word in self.substr_validation_words):
+            return 1
         else:
-            raise ValueError("model_provider must be either 'OpenAI' or 'Anthropic'")
+            return 0
+
+    @abstractmethod
+    def get_encoding(self, context):
+        pass
+
+    @abstractmethod
+    def get_decoding(self, encoded_context):
```

Comment:
I'm afraid this approach may not work for Gemini models, because we do not have public tokenizer for them.

Reply:
Although Gemini's tokenizer isn't public, our use case for tokenization—to identify fact placement depth—is effectively served by public tokenizers like 'tiktoken'. This ensures our test integrity without dependency on google's tokenizer.